lib/MusicBrainz/Server/Data/Utils.pm

```diff
@@ -328,8 +325,12 @@ sub sanitize {
     my $t = shift;
 
     $t = NFC($t);
+    # Before removing invalid characters, convert space control characters
+    # into U+0020 (or else they'll be removed).
+    $t = collapse_whitespace($t);
```

Comment:
Are we sure this doesn't remove anything we don't want to remove? (I guess we already remove full-width spaces anyway, although see MBS-5555

Also, we have a list of chars to avoid at https://musicbrainz.org/doc/Internationalization#Unicode_issues - not sure if we should block all of them while at it, but I guess dropping Ⅻ or whatnot would be weird. Maybe we should add a "split_roman_numbers" function here.

Reply:
It shouldn't remove anything we didn't already remove, AFAICT. (If you can think of an example where it might, I can add a test.)

Re: MBS-5555, IDEOGRAPHIC SPACE is still prohibited by the doc you linked, so we should probably make a decision on whether we want to allow that first. :) The others listed in that doc are all already stripped (or converted to U+0020 where appropriate) except for fullwidth latin and halfwidth kana/hangul, and roman numerals, but those don't make sense to remove in `sanitize` since they're valid printable characters. We'd probably want to add FormHandler validation for those instead to inform the user that they shouldn't be entered.